pkt-meta/src/dst_vni_lookup/mod.rs

```diff
@@ -186,3 +186,167 @@ impl<Buf: PacketBufferMut> NetworkFunction<Buf> for DstVniLookup {
+            .insert(Prefix::from("192.168.0.0/16"), dst_vni_100_192_168_0_0_16);
+        vni_table_100.dst_vnis.insert(
+            Prefix::from("::192.168.1.0/120"),
+            dst_vni_100_192_168_1_0_24,
+        );
+        vni_table_100.dst_vnis.insert(
+            Prefix::from("::192.168.0.0/112"),
+            dst_vni_100_192_168_0_0_16,
+        );
+
+        // VNI 100
+        let mut vni_table_200 = VniTable::new();
+        let dst_vni_200_192_168_2_0_24 = vni201;
+        let dst_vni_200_192_168_0_0_16 = vni202;
+        vni_table_200
+            .dst_vnis
+            .insert(Prefix::from("192.168.2.0/24"), dst_vni_200_192_168_2_0_24);
+        vni_table_200
+            .dst_vnis
+            .insert(Prefix::from("192.168.2.0/16"), dst_vni_200_192_168_0_0_16);
+        vni_table_200.dst_vnis.insert(
+            Prefix::from("::192.168.2.0/120"),
+            dst_vni_200_192_168_2_0_24,
+        );
+        vni_table_200.dst_vnis.insert(
+            Prefix::from("::192.168.0.0/112"),
+            dst_vni_200_192_168_0_0_16,
+        );
+
+        ////////////////////////////
+        // Setup VNI tables writer
+        let mut vni_tables = VniTables::new();
+        vni_tables.tables_by_vni.insert(vni100, vni_table_100);
+        vni_tables.tables_by_vni.insert(vni200, vni_table_200);
+        let mut vnitablesw = VniTablesWriter::new();
+        vnitablesw.update_vni_tables(vni_tables);
+
+        ////////////////////////////
+        // Setup DstVniLookup stage
+        let mut dst_vni_lookup = DstVniLookup::new("test", vnitablesw.get_reader());
+
+        ////////////////////////////
+        // Test IPv4 packets
+
+        let p_100_dst_addr_192_168_1_1 =
+            create_test_packet(Some(vni100), "192.168.1.1".parse().unwrap());
+        let p_100_dst_addr_192_168_100_1 =
+            create_test_packet(Some(vni100), "192.168.100.1".parse().unwrap());
+        let p_200_dst_addr_192_168_2_1 =
+            create_test_packet(Some(vni200), "192.168.2.1".parse().unwrap());
+        let p_200_dst_addr_10_0_0_1 = create_test_packet(Some(vni100), "10.0.0.1".parse().unwrap());
+
+        let packets_in = [
+            p_100_dst_addr_192_168_1_1,
+            p_100_dst_addr_192_168_100_1,
+            p_200_dst_addr_192_168_2_1,
+            p_200_dst_addr_10_0_0_1,
+        ];
+        let packets = dst_vni_lookup
+            .process(packets_in.into_iter())
+            .collect::<Vec<_>>();
+
+        assert_eq!(packets.len(), 4);
+        assert_eq!(packets[0].meta.dst_vni, Some(dst_vni_100_192_168_1_0_24));
+        assert!(!packets[0].is_done());
+        assert_eq!(packets[1].meta.dst_vni, Some(dst_vni_100_192_168_0_0_16));
+        assert!(!packets[1].is_done());
+        assert_eq!(packets[2].meta.dst_vni, Some(dst_vni_200_192_168_2_0_24));
+        assert!(!packets[2].is_done());
+        assert_eq!(packets[3].meta.dst_vni, None);
+        assert_eq!(packets[3].get_done(), Some(DoneReason::Unroutable));
```

Comment:
Maybe add a test case for when we can't find the source VNI for the first lookup, too?